src/main/java/com/example/springbootapp/exception/EmployeeExceptionHandler.java

```diff
@@ -33,6 +39,37 @@ public ResponseError employeeNotSaved(EmployeeNotSavedException exception) {
         return new ResponseError(HttpStatus.BAD_REQUEST, exception.getMessage());
     }
 
+    @ResponseBody
+    @ExceptionHandler(ConstraintViolationException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ValidationResponseError onConstraintValidationException(
+            ConstraintViolationException exception
+    ) {
+        LOGGER.error(exception.getMessage(), exception);
+        final List<Violation> violations = exception.getConstraintViolations().stream()
```

Comment:
Extract this logic in a separate private method please. The same for the onMethodArgumentNotValidException method.